mk/spksrc.cross-rust-env.mk

```diff
@@ -0,0 +1,62 @@
+# Configuration for rust build
+# 
+
+RUST_TOOLCHAIN ?= stable
+
+RUST_TARGET =
+# map archs to rust targets
+ifeq ($(findstring $(ARCH), $(x64_ARCHS)),$(ARCH))
+RUST_TARGET = x86_64-unknown-linux-gnu
+endif
+ifeq ($(findstring $(ARCH), $(i686_ARCHS)),$(ARCH))
+RUST_TARGET = i686-unknown-linux-gnu
+endif
+ifeq ($(findstring $(ARCH), $(ARMv5_ARCHS)),$(ARCH))
+# may be not supported for cargo
+RUST_TARGET = armv5te-unknown-linux-gnueabi
+endif
+ifeq ($(findstring $(ARCH), $(ARMv7_ARCHS)),$(ARCH))
+RUST_TARGET = armv7-unknown-linux-gnueabihf
+endif
+ifeq ($(findstring $(ARCH), $(ARMv7L_ARCHS)),$(ARCH))
+RUST_TARGET = armv7-unknown-linux-gnueabi
+endif
+ifeq ($(findstring $(ARCH), $(ARMv8_ARCHS)),$(ARCH))
+RUST_TARGET = aarch64-unknown-linux-gnu
+endif
+ifeq ($(findstring $(ARCH), $(PPC_ARCHS)),$(ARCH))
+RUST_TARGET = powerpc-unknown-linux-gnu
+endif
+ifeq ($(RUST_TARGET),)
+$(error Arch $(ARCH) not supported)
+endif
+
+# Use distrib folder as cargo download cache
+CARGO_HOME_PATH = /spksrc/distrib/cargo
+ENV += CARGO_HOME=$(CARGO_HOME_PATH)
+ENV += PATH=:$(CARGO_HOME_PATH)/bin/:$(PATH)
+
+ifeq (,$(shell $(ENV) which rustup))
+install_rustup:
+	@echo "  ==> install rustup" ; \
+	curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | \
+	        CARGO_HOME=$(CARGO_HOME_PATH) sh -s -- -y
+else
+install_rustup:
+	@echo "  ==> rustup already installed" ;
+endif
+
+install_rust_toolchain: install_rustup
+	@echo "  ==> install rust toolchain [$(RUST_TOOLCHAIN)]" ; \
+	env $(ENV) rustup toolchain install $(RUST_TOOLCHAIN) ;
+	env $(ENV) rustup default $(RUST_TOOLCHAIN) ;
+
+# Install rust target on demand:
+install_rust_target: install_rust_toolchain
+	@echo "  ==> install rust target [$(RUST_TARGET)]" ; \
+	env $(ENV) rustup target install $(RUST_TARGET) ;
+
+# Set linker environment variable
+RUST_LINKER_ENV = CARGO_TARGET_$(shell echo $(RUST_TARGET) | tr - _ | tr a-z A-Z)_LINKER
+ENV += $(RUST_LINKER_ENV)=$(TC_PATH)$(TC_PREFIX)gcc
+ENV += CARGO_BUILD_TARGET=$(RUST_TARGET)
```

Comment:
Missing is defining `RUSTC` environment variable and adding it up to the tc_vars.mkl file.  Also including a `RUSTFLAGS`, worst case empty at first.